packages/firebase_admob/android/src/main/java/io/flutter/plugins/firebaseadmob/FirebaseAdMobPlugin.java

```diff
@@ -24,6 +27,12 @@
 
   RewardedVideoAdWrapper rewardedWrapper;
 
+  static Function<UnifiedNativeAd, View> nativeAdGenerator;
```

Comment:
Is it reasonable for an app to have multiple Native Ad factories? (e.g the same app wants to potentially display native ads in multiple ways?)

Reply:
Couldn't the factory handle this situation? Is there more information that the factory would need to display Native Ads differently?

Although, this does highlight that the creation of the Ad is done on the platform side while the placement is set on the dart side (e.g. anchoring the ad to the top or bottom). The user might want to place the ad based on its content.

A simple solution would be to pass the `UnifiedNativeAd` data to the user on the dart side so the platform and dart sides will sync on how to place and create the ad. This could be a separate PR though after the initial basic implementation for Android and iOS.

Reply:
Interesting, I'm probably missing something, how would this work for an app with 2 ads on the same screen, each should be styled differently (say one is presented over a light background and the other is presented over a dark background).

When this factory is invoked, how does it know whether it should style for the dark background or the light background?

Reply:
There are [NativeAdOptions](https://developers.google.com/android/reference/com/google/android/gms/ads/formats/NativeAdOptions) for NativeAds, so we could add a parameter in `NativeAd` that passes custom arguments. Similar to `PlatformViewFactory.`

I would add this in a different PR after iOS is implemented.

Reply:
NativeAdOptions seems to be a very constrained object, how would you e.g express light/dark background using it?

Reply:
Discussed offline:

We'll include a `Map<String, dynamic> customOptions` parameter in the `NativeAdFactory. createNativeAd` method.